### PR TITLE
Update SC dialog to save/read file structure to/from schema compare file

### LINKED ISSUE #20332: Save scmp file doesn't store "File Structure" information for database projects as target
Steps to Reproduce:

1. Open schema compare.
2. Update source and target information for comparison, with target set as Database projects.
3.  Save scmp file.

Expected behavior: All the provided information is stored in the file.
Actual: It doesn't store "File structure" information.

### PR DESCRIPTION
**Pull request by SakshiS-harma**
This PR fixes #20332.
With latest STS changes (https://github.com/microsoft/sqltoolsservice/pull/1878 and https://github.com/microsoft/sqltoolsservice/pull/1952), user can now save file structure into the schema compare file and read the stored information from the scmp file and present it on the schema compare dialog.

scmpBug

### CODE REVIEW

**Comment by SakshiS-harma** [comment]
Found this bug while testing: https://github.com/microsoft/azuredatastudio/issues/22728

File: extensions/schema-compare/src/schemaCompareMainWindow.ts [hunk 1]
```diff
@@ -1151,8 +1151,8 @@ export class SchemaCompareMainWindow {
 				connectionDetails: undefined,
 				projectFilePath: endpoint.projectFilePath,
 				targetScripts: [],
-				dataSchemaProvider: '',
-				folderStructure: mssql.ExtractTarget.schemaObjectType			// TODO: Pick this automatically from the scmp file, after issue #20332 is resolved (check dsp as well)
+				dataSchemaProvider: endpoint.dataSchemaProvider,
+				extractTarget: endpoint.extractTarget			// TODO: Pick this automatically from the scmp file, after issue #20332 is resolved (check dsp as well)
```
Comment (Benjin) [inline extensions/schema-compare/src/schemaCompareMainWindow.ts:1155]: Is this PR resolving this `TODO`/can the comment be removed?

**Review by Benjin — APPROVED** [review]

**Review by kisantia — APPROVED** [review]
Also bump the azdata dependency to 1.44 in schema compare's package.json since this depends on changes made in mssql

File: extensions/schema-compare/src/schemaCompareMainWindow.ts (hunk 1 above)
Comment (SakshiS-harma) [reply extensions/schema-compare/src/schemaCompareMainWindow.ts]: My bad, this PR addresses this itself and it completely slipped off ! Fixed it.

**Comment by SakshiS-harma** [comment]
> Also bump the azdata dependency to 1.44 in schema compare's package.json since this depends on changes made in mssql

Done.

**Pull request merged**